lib/compress.js

```diff
@@ -774,13 +767,6 @@ merge(Compressor.prototype, {
                 }
             }
 
-            // Remove extraneous empty statments in block after removing var definitions.
-            // Leave at least one statement in `statements`.
```

Comment:
```
var scope = compressor.find_parent(AST_Scope);
```
Interesting. So that's probably why the harmony branch has the hack related to scope.